samples/geo_vertex_format.html

```diff
@@ -0,0 +1,478 @@
+            return;
+        }
+
+        // -- Init program
+        var program = createProgram(gl, getShaderSource('vs'), getShaderSource('fs'));
+        var unifModel = gl.getUniformLocation(program, 'u_model');
+        var unifModelInvTrans = gl.getUniformLocation(program, 'u_modelInvTrans');
+        var unifViewProj = gl.getUniformLocation(program, 'u_viewProj');
+        var unifLightPosition = gl.getUniformLocation(program, 'u_lightPosition');
+
+        var unifTex2D = gl.getUniformLocation(program, 's_tex2D');
+        var unifAmbient = gl.getUniformLocation(program, 'u_ambient');
+
+        // -- Init geometries
+        var positions = new Float32Array([
+          // Front face
+          -1.0, -1.0,  1.0,
+           1.0, -1.0,  1.0,
+           1.0,  1.0,  1.0,
+          -1.0,  1.0,  1.0,
+
+          // Back face
+          -1.0, -1.0, -1.0,
+          -1.0,  1.0, -1.0,
+           1.0,  1.0, -1.0,
+           1.0, -1.0, -1.0,
+
+          // Top face
+          -1.0,  1.0, -1.0,
+          -1.0,  1.0,  1.0,
+           1.0,  1.0,  1.0,
+           1.0,  1.0, -1.0,
+
+          // Bottom face
+          -1.0, -1.0, -1.0,
+           1.0, -1.0, -1.0,
+           1.0, -1.0,  1.0,
+          -1.0, -1.0,  1.0,
+
+          // Right face
+           1.0, -1.0, -1.0,
+           1.0,  1.0, -1.0,
+           1.0,  1.0,  1.0,
+           1.0, -1.0,  1.0,
+
+          // Left face
+          -1.0, -1.0, -1.0,
+          -1.0, -1.0,  1.0,
+          -1.0,  1.0,  1.0,
+          -1.0,  1.0, -1.0
+        ]);
+        var vertexPosBuffer = gl.createBuffer();
+        gl.bindBuffer(gl.ARRAY_BUFFER, vertexPosBuffer);
+        gl.bufferData(gl.ARRAY_BUFFER, positions, gl.STATIC_DRAW);
+        gl.bindBuffer(gl.ARRAY_BUFFER, null);
+
+        // http://stackoverflow.com/questions/28688838/convert-a-number-into-a-16-bit-float-stored-as-bytes-and-back
+        // http://croquetweak.blogspot.de/2014/08/deconstructing-floats-frexp-and-ldexp.html
+        function frexp(value) {
+            if (value === 0) return [value, 0];
+            var data = new DataView(new ArrayBuffer(8));
+            data.setFloat64(0, value);
+            var bits = (data.getUint32(0) >>> 20) & 0x7FF;
+            if (bits === 0) {
+                data.setFloat64(0, value * Math.pow(2, 64));
+                bits = ((data.getUint32(0) >>> 20) & 0x7FF) - 64;
+            }
+            var exponent = bits - 1022,
+                mantissa = ldexp(value, -exponent);
+            return [mantissa, exponent];
+        }
+        
+        function ldexp(mantissa, exponent) {
+            // avoid multiplying by infinity and zero
+            return exponent > 1023 ? mantissa * Math.pow(2, 1023) * Math.pow(2, exponent - 1023) : 
+                exponent < -1074 ? mantissa * Math.pow(2, -1074) * Math.pow(2, exponent + 1074) : mantissa * Math.pow(2, exponent);
+        }
+        
+        function encodeFloat16(value) {
```

Comment:
This function has a lot of allocations that we can avoid by using file-scoped variables through closures instead of function scoped variables.

Also, let's move this to a separate .js file for now, and update #91 with a note to better test and package this (we'll want to write careful fine-grained unit tests).